server/controllers/payment.js

```diff
@@ -1,65 +1,131 @@
   if (id === req.params.id) {
-    stripePayment(req, res, function (err) {
-      if (err) {
-        return res.json({
-          success: false,
-          errors: {
-            title: "Stripe payment failed",
-            detail: err.message,
-            error: err,
-          },
-        });
-      }
+    await stripePayment(req, res);
 
-      res
-        .status(200)
-        .send({ message: "The payment session has been successfully created" });
-    });
+    res
+      .status(200)
+      .send({ message: "The payment session has been successfully created", url: req.url });
   }
 });
 
-// @route POST /:id/create-customer
+// @route GET /:id/create-customer
 // @desc create customer
 // @access Private
-exports.createCustomer = asyncHandler(async (req, res, next) => {
+exports.createCustomer = asyncHandler(async (req, res) => {
   const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
+
   if (id === req.params.id) {
-    createCustomer(req, res, function (err) {
-      if (err) {
-        return res.json({
-          success: false,
-          errors: {
-            title: "Create customer failed",
-            detail: err.message,
-            error: err,
-          },
-        });
-      }
+    await createCustomer(req, res);
+
+    res
+      .status(200)
+      .send({ message: "The customer has been successfully created", customer: req.createdCustomer });
+  }
+});
+
+// @route GET /:id/new-customer-create-intent
+// @desc create customer and intent
+// @access Private
+exports.newCustomerCreateIntent = asyncHandler(async (req, res) => {
+  const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
 
+  if (id === req.params.id) {
+    await createCustomer(req, res);
+
+    await createIntent(req, res);
+
+    res
+      .status(200)
+      .send({
+        message: "Both customer and intent have been successfully created",
+        customer: req.createdCustomer,
+        intent: req.intent
+      });
+  }
+});
+
+// @route GET /:id/add-card
+// @desc get or create customer and create intent
+// @access Private
+exports.getOrCreateCustomerCreateIntent = asyncHandler(async (req, res) => {
+  const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
+
+  if (id === req.params.id) {
+    await getCustomer(req, res);
+
+    if (req.createdCustomer === undefined) {
+      await createCustomer(req, res);
+    }
+
+    await createIntent(req, res);
+
+    res
+      .status(200)
+      .send({
+        message: "Both customer and intent have been successfully created",
+        customer: req.createdCustomer,
+        intent: req.intent
+      });
+  }
+});
+
+// @route GET /:id/all-payment-methods
+// @desc get all payment methods
+// @access Private
+exports.getAllPaymentMethods = asyncHandler(async (req, res) => {
+  const id = req.user.id;
+  const user = await User.findById(id);
+  req.email = user.email;
```

Comment:
Not sure if this is required

Reply:
Yes, it needs because req.email is a parameter of Stripe createCustomer function.